### PR TITLE
MERGE CREATE WEB SOCKET ADAPTER FOR CLEAN START:

### LINKED ISSUE #54: Create web socket adapter (also called Gateway) for clean start
This day we start the implementation of web socket for the game engine.
We was almost done the job but we was stuck and we was playing against the clock.
Anyway, I submit an issue for having a clean start.
I take the code from nb/multiplayer_ball and start debug whats wrong. 
More explanation on the PR come with comment on what we was doing.
Nothing except connection will be done for this issue. So the code will not have to many diff from original sources code.

### PR DESCRIPTION
**Pull request by apayetRun**
close #54 
### Issue  description:
This day we start the implementation of web socket for the game engine.
We was almost done the job but we was stuck and we was playing against the clock.
Anyway, I submit an issue for having a clean start.
I take the code from nb/multiplayer_ball and start debug whats wrong.
More explanation on the PR come with comment on what we was doing.
Nothing except connection will be done for this issue. So the code will not have to many diff from original sources code.

### CODE REVIEW

File: backend/app/src/app.module.ts [hunk 1]
```diff
@@ -11,11 +11,16 @@ import
...
+import { GameSocketModule } from "./game-socket/game-socket.module";
 
 
 @Module(
 {
-	imports: [AdminsModule],
+	imports:
+	[
+		AdminsModule,
+		GameSocketModule
```
Comment (apayetRun) [inline backend/app/src/app.module.ts:22]: Issue solve: we forgot to import the GameSocketModule, that's why :)

File: backend/app/src/game-socket/GameSocketEvents.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { Server, Socket } from "socket.io";
+
+
+@WebSocketGateway(
+{
+	cors:
+	{
+		origin: "*"
+	},
+	// namespace: "/game-engine-v2d"
```
Comment (apayetRun) [inline backend/app/src/game-socket/GameSocketEvents.ts:19]: This one is commented, for now, not in the scope of the issue. 
Meeting requiered.

File: backend/app/src/game-socket/GameSocketEvents.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+})
+export class GameSocketEvents
+	implements OnGatewayInit, OnGatewayConnection, OnGatewayDisconnect
```
Comment (apayetRun) [inline backend/app/src/game-socket/GameSocketEvents.ts:22]: We forgot to implement the class lifecycle decorator that handle respectively afterInit, handleConnection, and handleDisconnect

File: front/app/src/Views/TestBall/TestBall.tsx [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+		socket.on("disconnect", disconnect);
+		socket.on("error", connectError);
+		socket.connect();
+		return (() =>
+		{
+			socket.off("connect", connect);
+			socket.off("disconnect", disconnect);
+			socket.off("error", connectError);
+		});
+	}, []);
```
Comment (apayetRun) [inline front/app/src/Views/TestBall/TestBall.tsx:113]: Nota Bene : the empty array : [] is intentionnaly here, there means  no dependency to the effect.
The useEffect is setted like a lifecycle function 
In this configuration, the use effect is called only one time after the component is mounted.

File: front/app/src/Views/TestBall/TestBall.tsx [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+		const	connectError = (error: Error) =>
+		{
+			console.error("ws_connect_error", error);
+		};
+
+		socket.on("connect", connect);
+		socket.on("disconnect", disconnect);
+		socket.on("error", connectError);
+		socket.connect();
+		return (() =>
```
Comment (apayetRun) [inline front/app/src/Views/TestBall/TestBall.tsx:107]: Safely unmount the component

**Review by apayetRun — COMMENTED** [review]
I comment all modification, tell me if question's persist.

**Review by nicolkaver — APPROVED** [review]
Thank you for your work and your comments ! Looking forward to continuing our work this morning.

**Review by nicolkaver — APPROVED** [review]
Thank you for the job !

**Review by Keeghaan — APPROVED** [review]
nice job !

**Pull request merged**